front_end/panels/timeline/TimelineUIUtils.ts

```diff
@@ -1708,6 +1729,27 @@ export class TimelineUIUtils {
     return contentHelper.fragment;
   }
 
+  /**
+   * Get attribution data for a given URL.
+   *
+   * @param {string} url URL to check for attribution data.
+   * @param {Trace.Types.Events.PerformanceAttribution[]} attributions Array of performance attributions. Immutable.
+   *
+   * @return {string|null} Attribution name or null if no attribution is found.
+   */
+  static getAttributionForUrl(url: string, attributions: Trace.Types.Events.PerformanceAttribution[]): string|null {
```

Comment:
Q: do we really need to pass the attributions as a parameter? Try accessing parsedTrace here so we don't need to.